lib/eve.js

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect } from 'preact/hooks'
+
+// throttle a function call to avoid calling it too much
+export const throttle = (fn, delay, t, call = () => ((t = 0), fn())) => () =>
+  t || (t = setTimeout(call, delay))
+
+// create an observable or event
+export const eve = (...args) => {
```

Comment:
This is a event like construct, that allow to write reactive code, that is, code that's subscribe to value changes and react to it.

This is useful to have loosely coupled element, each elements can define how they want to react without carring about interacting with others or changing code else where.